mempool/cat/pool.go

```diff
@@ -168,7 +168,7 @@ func (txmp *TxPool) Height() int64 {
 
 // Has returns true if the transaction is currently in the mempool
 func (txmp *TxPool) Has(txKey types.TxKey) bool {
-	return txmp.store.has(txKey)
+	return txmp.store.has(txKey) || txmp.store.isReserved(txKey)
 }
```

Comment:
[discussion] As a caller of mempool methods, the divergence in return values between `.Has()` and `.Get()` seems weird. Assume `txKey=foo` is in the reserved map.

```go
has := txmp.Has(foo)
assert.True(t, has)

tx, ok := txmp.Get(foo) 
assert.True(t, ok) // fails because foo isn't in the store, it's in the reserved map
```

Do we need to support both methods?

Reply:
hmmmmmmmmmmmmmmmmm good point

Reply:
Yeah I don't think we should could `reserved` transasctions as being in the mempool. There should be another way to fix this bug. I can have a look at it at a later point